File: frontend/src/features/orderManagement/order.service.js
import apiClient from "../../services/apiClient";

/**
 * GET /v1/orders — Admin/Manager order history with filters
 * @param {{ page?: number, size?: number, status?: string, branchId?: number, fromDate?: string, toDate?: string }} params
 */
export const getOrders = async (params) => {
  const res = await apiClient.get("/api/v1/orders", { params });
  return res.data.result;
};

/**
 * GET /v1/orders/:id — Order detail (CUSTOMER, STORE_MANAGER, FRANCHISE_ADMIN)
 */
export const getOrderDetail = async (id) => {
  const res = await apiClient.get(`/api/v1/orders/${id}`);
  return res.data.result;
};

/**
 * GET /v1/orders/:id/status — Order status (CUSTOMER, FRANCHISE_ADMIN, STORE_MANAGER)
 */
export const getOrderStatus = async (id) => {
  const res = await apiClient.get(`/api/v1/orders/${id}/status`);
  return res.data.result;
};

/**
 * PATCH /v1/orders/:id/cancel — Cancel an order
 */
export const cancelOrder = async (id) => {
  const res = await apiClient.patch(`/api/v1/orders/${id}/cancel`);
  return res.data.result;
};

File: order-service/product-order-payment/src/main/java/fsoft/franchise/service/OrderService.java
package fsoft.franchise.service;

import fsoft.franchise.dto.orders.*;
// import fsoft.franchise.model.dto.*;
import org.springframework.stereotype.Service;

import fsoft.franchise.enums.OrderStatus;
import java.util.List;
import java.util.UUID;
import java.time.LocalDate;
import java.util.Optional;

@Service
public interface OrderService {

    public OrderCancelResponse cancelOrder(UUID orderId, UUID userId, String role);

    public OrderDetailResponse getOrderDetail(String orderId, UUID currentUserId, String role);

    /**
     * Get order status. Customer chỉ xem được order của mình; ADMIN và
     * MANAGER xem được mọi order.
     */
    OrderStatusResponse getStatus(UUID orderId, UUID userId, String role);

    /**
     * Order History: pagination + filter. Chỉ ADMIN và MANAGER được
     * gọi.
     */
    OrderHistoryPage getOrderHistory(int page, int size,
            Optional<String> status,
            Optional<UUID> storeId,
            Optional<LocalDate> fromDate,
            Optional<LocalDate> toDate,
            String role);

    /**
     * Get order history for the authenticated customer (their own orders only).
     */
    OrderHistoryPage getMyOrders(UUID customerId, int page, int size);

    /** Returns all possible order statuses. */
    List<OrderStatus> getOrderStatuses();

    /** Returns all relevant enums for order creation/management. */
    OrderEnumsResponse getOrderEnums();

    CreateOrderResponse createOrder(CreateOrderRequest request, UUID customerId);


    EstimateResponse estimatePreparationTime(UUID storeId, int itemCount);

    FlagOrderResponse flagOrder(UUID orderId, FlagOrderRequest request, UUID currentUserId, String role,
            Long currentUserStoreId);

    CreatePosOrderResponse createPosOrder(CreatePosOrderRequest request, UUID staffId);
    UpdatePosOrderResponse updatePosOrder(UUID orderId, UpdatePosOrderRequest request, UUID staffId);
}
